app/gui/src/integration_test.rs

```diff
@@ -0,0 +1,155 @@
+//! Enso testing fixtures
+//!
+//! Support structures for high-level testing crates that must connect to the Project Manager.
+//! This includes integration tests and batch-mode application profiling.
+
+use crate::prelude::*;
+use enso_web::traits::*;
+
+use crate::executor::web::EventLoopExecutor;
+use crate::initializer::setup_global_executor;
+use crate::Ide;
+use enso_frp::future::EventOutputExt;
+use enso_web::Closure;
+use enso_web::HtmlDivElement;
+use ensogl::application::test_utils::ApplicationExt;
+use std::pin::Pin;
+
+
+
+/// Reexports of commonly-used structures, methods and traits.
+pub mod prelude {
+    pub use super::Fixture;
+
+    pub use super::wait_a_frame;
+    pub use crate::prelude::*;
+    pub use enso_frp::future::EventOutputExt;
+}
+
+
+
+// =======================
+// === IntegrationTest ===
+// =======================
+
+/// A fixture for each IDE integration tests. During setup, the executor and [`Ide`] structure are
+/// initialized.
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct Fixture {
+    pub executor: EventLoopExecutor,
+    pub ide:      Ide,
+    pub root_div: HtmlDivElement,
+}
+
+impl Fixture {
+    /// Initializes the executor and `Ide` structure and returns new Fixture.
+    pub async fn new(open_to_project_view: bool) -> Self {
```

Comment:
I don't understand the arg `open_to_project_view` - what it is responsible for?

Reply:
Refactored this to an enum to make the two alternatives clear.